### ISSUE TITLE
No schema available in Grapiql after upgrade to 1.6.1

### ISSUE THREAD

**Issue by ohuu** [wp-graphql/wp-graphql#2049]
Title says it all really

**Comment by jasonbahl** [comment]
@PaperPlaneSoftware v1.6 introduced Lazy Type Loading, and some extensions need to make some upgrades to play nice with this. If you're using WPGraphQL for WooCommerce or WPGraphQL for Gravity Forms (possibly others) you might need to wait until they're compatible with the Lazy Type loading to upgrade. 

If you're not using any extensions or have any code you've added to extend WPGraphQL yourself, and you're still having this issue, can you share more details about what plugins you do have active? what WordPress version? What PHP version, etc? 

The more info possible to help us reproduce, the better. 

I'm on v1.6.1 and using GraphiQL myself, so the more info I have to reproduce your environment, the better.

**Comment by PaperPlaneSoftware** [comment]
Thanks @jasonbahl. Here's a list of WPGraphQL plugins I'm using: Woocommerce, JWT and ACF

**Comment by jasonbahl** [comment]
@PaperPlaneSoftware could you maybe share your ACF Field Group Export with me? You could do it here, or in Slack via DM. 

Also, do you have ACF Options Pages exposed in WPGraphQL? I think we just discovered a bug related to that situation here: https://github.com/wp-graphql/wp-graphql/issues/2047#issuecomment-891182361

**Comment by eihsan94** [comment]
Im also experiencing the same issue,

**What did I do**
updating WPGraphQL from 1.5.5 to 1.6.1

**What IS My Infra**
- headless WP with graphql
- Next js frontend


**What Happen** 
1. the graphql schema stop working(The schema stop showing in IDE)
![image](https://user-images.githubusercontent.com/26185320/128119753-0863e471-5c8b-41be-b2da-b49f87828422.png)
2. in vercel codegen cant found the schema in wordpres.ts types
![image](https://user-images.githubusercontent.com/26185320/128120062-15bcbc62-72f2-4efe-9b5b-1e17b5589580.png)
3. Console log
![image](https://user-images.githubusercontent.com/26185320/128122907-ab33aea7-5ab3-4198-a87c-a29bed04bbbd.png)


**Does this work in previous version** 
YES IT WORKS IN v1.5.5


**this is my Snippets** 

`{"generator":"Code Snippets v2.14.1","date_created":"2021-05-11 05:24","snippets":[{"name":"block_register_prd","scope":"admin","code":"function block_register_prd() {\nwp_enqueue_script( 'gutenberg-examples-01', 'https:\/\/beyond-blocks-prd-hosting-bucket.s3-ap-northeast-1.amazonaws.com\/index.js', array( 'wp-blocks', 'wp-element' ), '1.0.0', true );\n}\nadd_action( 'enqueue_block_editor_assets', 'block_register_prd' );","priority":"10"}]}`

`{"generator":"Code Snippets v2.14.0","date_created":"2021-04-20 08:52","snippets":[{"name":"Expand Editor Capability","scope":"admin","code":"$role = get_role( 'editor' );\n$role->add_cap( 'edit_users' );\n$role->add_cap( 'create_users' );\n$role->add_cap( 'manage_network_users' );\n$role->add_cap( 'list_users' );\n$role->add_cap( 'promote_users' );\n","priority":"10"}]}`

`{"generator":"Code Snippets v2.14.0","date_created":"2021-04-20 10:14","snippets":[{"name":"Extend User Profile Form","scope":"admin","code":"\/\/ hooks at updating user profile\nfunction edit_additional_profile ($user_id) {\n\tif ( isset( $_POST['user_image'] ) ) {\n        update_user_meta($user_id, 'user_image', $_POST['user_image']);\n    }\n\tif ( isset( $_POST['user_title'] ) ) {\n        update_user_meta($user_id, 'user_title', $_POST['user_title']);\n    }\n\tif ( isset( $_POST['user_website'] ) ) {\n        update_user_meta($user_id, 'user_website', $_POST['user_website']);\n    }\n\tif ( isset( $_POST['user_twitter'] ) ) {\n        update_user_meta($user_id, 'user_twitter', $_POST['user_twitter']);\n    }\n\tif ( isset( $_POST['user_facebook'] ) ) {\n        update_user_meta($user_id, 'user_facebook', $_POST['user_facebook']);\n    }\n\tif ( isset( $_POST['user_instagram'] ) ) {\n        update_user_meta($user_id, 'user_instagram', $_POST['user_instagram']);\n    }\n}\nadd_action('profile_update', 'edit_additional_profile'); \nfunction append_custom_fields ($user) {\n\t?>\n\t<table id=\"custom_user_field_table\" class=\"form-table\">\n        <tr id=\"custom_user_field_row\">\n            <th>\n                <label for=\"user_image\">Profile Image URL<\/label>\n            <\/th>\n            <td>\n                <input type=\"text\" name=\"user_image\" id=\"user_image\" value=\"<?php echo esc_attr( get_the_author_meta( 'user_image', $user->ID ) ); ?>\" class=\"regular-text\" \/><br \/>\n                <span class=\"description\">Media\u3067\u30a2\u30c3\u30d7\u30ed\u30fc\u30c9\u3057\u305f\u753b\u50cf\u306e\u30ea\u30f3\u30af\u3092\u5165\u529b\u3057\u3066\u304f\u3060\u3055\u3044<\/span>\n            <\/td>\n        <\/tr>\n\t\t<tr id=\"custom_user_field_row\">\n            <th>\n                <label for=\"user_title\">User Title<\/label>\n            <\/th>\n            <td>\n                <input type=\"text\" name=\"user_title\" id=\"user_title\" value=\"<?php echo esc_attr( get_the_author_meta( 'user_title', $user->ID ) ); ?>\" class=\"regular-text\" \/><br \/>\n                <span class=\"description\">\u4f8b: \u30e9\u30a4\u30bf\u30fc\u30fb\u30b3\u30e9\u30e0\u30cb\u30b9\u30c8\u306a\u3069\u306e\u80a9\u66f8\u3092\u5165\u529b\u3057\u3066\u304f\u3060\u3055\u3044<\/span>\n            <\/td>\n        <\/tr>\n\t\t<tr id=\"custom_user_field_row\">\n            <th>\n                <label for=\"user_website\">Website<\/label>\n            <\/th>\n            <td>\n                <input type=\"text\" name=\"user_website\" id=\"user_website\" value=\"<?php echo esc_attr( get_the_author_meta( 'user_website', $user->ID ) ); ?>\" class=\"regular-text\" \/><br \/>\n                <span class=\"description\">\u30e6\u30fc\u30b6\u30fc\u306eWebsite\u3092\u5165\u529b\u3057\u3066\u304f\u3060\u3055\u3044<\/span>\n            <\/td>\n        <\/tr>\n\t\t<tr id=\"custom_user_field_row\">\n            <th>\n                <label for=\"user_twitter\">Twitter<\/label>\n            <\/th>\n            <td>\n                <input type=\"text\" name=\"user_twitter\" id=\"user_twitter\" value=\"<?php echo esc_attr( get_the_author_meta( 'user_twitter', $user->ID ) ); ?>\" class=\"regular-text\" \/><br \/>\n                <span class=\"description\">Twitter\u306e\u30a2\u30ab\u30a6\u30f3\u30c8\u306e@\u4ee5\u4e0b\u3092\u5165\u529b\u3057\u3066\u304f\u3060\u3055\u3044<\/span>\n            <\/td>\n        <\/tr>\n\t\t<tr id=\"custom_user_field_row\">\n            <th>\n                <label for=\"user_facebook\">Facebook<\/label>\n            <\/th>\n            <td>\n                <input type=\"text\" name=\"user_facebook\" id=\"user_facebook\" value=\"<?php echo esc_attr( get_the_author_meta( 'user_facebook', $user->ID ) ); ?>\" class=\"regular-text\" \/><br \/>\n                <span class=\"description\">Facebook\u306e\u30e6\u30fc\u30b6\u30fcID\u3092\u5165\u529b\u3057\u3066\u304f\u3060\u3055\u3044<\/span>\n            <\/td>\n        <\/tr>\n\t\t<tr id=\"custom_user_field_row\">\n            <th>\n                <label for=\"user_instagram\">Instagram<\/label>\n            <\/th>\n            <td>\n                <input type=\"text\" name=\"user_instagram\" id=\"user_instagram\" value=\"<?php echo esc_attr( get_the_author_meta( 'user_instagram', $user->ID ) ); ?>\" class=\"regular-text\" \/><br \/>\n                <span class=\"description\">Instagram\u306e\u30e6\u30fc\u30b6\u30fcID\u3092\u5165\u529b\u3057\u3066\u304f\u3060\u3055\u3044<\/span>\n            <\/td>\n        <\/tr>\n    <\/table>\n<?php\n}\nadd_action( 'show_user_profile', 'append_custom_fields' );\nadd_action( 'edit_user_profile', 'append_custom_fields' );\n","priority":"10"}]}`

`{"generator":"Code Snippets v2.14.0","date_created":"2021-04-20 10:14","snippets":[{"name":"Extend WPGraphQL schema","scope":"global","code":"function resolveUserImage($source) {\n\t$fields = get_user_meta($source->userId, 'user_image', true);\n\tif ( isset( $fields ) ) {\n      return $fields;\n    }\n    return null;\n}\nfunction resolveUserTitle($source) {\n\t$fields = get_user_meta($source->userId, 'user_title', true);\n\tif ( isset( $fields ) ) {\n      return $fields;\n    }\n    return null;\n}\nfunction resolveUserWebsite($source) {\n\t$fields = get_user_meta($source->userId, 'user_website', true);\n\tif ( isset( $fields ) ) {\n      return $fields;\n    }\n    return null;\n}\nfunction resolveUserTwitter($source) {\n\t$fields = get_user_meta($source->userId, 'user_twitter', true);\n\tif ( isset( $fields ) ) {\n      return $fields;\n    }\n    return null;\n}\nfunction resolveUserFacebook($source) {\n\t$fields = get_user_meta($source->userId, 'user_facebook', true);\n\tif ( isset( $fields ) ) {\n      return $fields;\n    }\n    return null;\n}\nfunction resolveUserInstagram($source) {\n\t$fields = get_user_meta($source->userId, 'user_instagram', true);\n\tif ( isset( $fields ) ) {\n      return $fields;\n    }\n    return null;\n}\nfunction add_custom_fields($fieldName, $resolver) {\n\t$field_config = [\n    'type' => 'String',\n    'args' => [],\n    'resolve' => $resolver\n  ];\n  register_graphql_field('user', $fieldName, $field_config);\n}\n\nadd_action( 'graphql_register_types', function () {\n\tadd_custom_fields('userImage', 'resolveUserImage');\n\tadd_custom_fields('userTitle', 'resolveUserTitle');\n\tadd_custom_fields('userWebsite', 'resolveUserWebsite');\n\tadd_custom_fields('userTwitter', 'resolveUserTwitter');\n\tadd_custom_fields('userFacebook', 'resolveUserFacebook');\n\tadd_custom_fields('userInstagram', 'resolveUserInstagram');\n});\n","priority":"10"}]}`


`{"generator":"Code Snippets v2.14.1","date_created":"2021-05-24 01:26","snippets":[{"name":"svg upload media support","scope":"global","code":"\/\/ Allow SVG\nadd_filter( 'wp_check_filetype_and_ext', function($data, $file, $filename, $mimes) {\n\n  global $wp_version;\n  if ( $wp_version !== '4.7.1' ) {\n     return $data;\n  }\n\n  $filetype = wp_check_filetype( $filename, $mimes );\n\n  return [\n      'ext'             => $filetype['ext'],\n      'type'            => $filetype['type'],\n      'proper_filename' => $data['proper_filename']\n  ];\n\n}, 10, 4 );\n\nfunction cc_mime_types( $mimes ){\n  $mimes['svg'] = 'image\/svg+xml';\n  return $mimes;\n}\nadd_filter( 'upload_mimes', 'cc_mime_types' );\n\nfunction fix_svg() {\n  echo '<style type=\"text\/css\">\n        .attachment-266x266, .thumbnail img {\n             width: 100% !important;\n             height: auto !important;\n        }\n        <\/style>';\n}\nadd_action( 'admin_head', 'fix_svg' );\n\n","priority":"10"}]}`

**Comment by jordanmaslyn** [comment]
Can confirm that WP GraphQL Gutenberg breaks the schema loading in GraphiQL for WPGraphQL v1.6.1

**Comment by eihsan94** [comment]
@jordanmaslyn @jasonbahl do u guys know any fix?

**Comment by stevepepple** [comment]
We also observed this issue. Have rolled back to version `1.5.9` for now. 

Plugins: 
- ACF to REST API
- Advanced Custom Fields PRO
- Custom Post Type UI
- WP Gatsby
- WP GraphQL Gutenberg
- WPGraphQL for Advanced Custom Fields
- Yoast SEO

**Comment by jasonbahl** [comment]
Thanks for the info! 

I have 2 open PRs (#2051, #2055) that address some known regressions caused by 1.6/1.6.1. 

I plan to get them released as 1.6.2 shortly and would love it if ya'll could test again once that's released 🙏🏻 

There might need to be some work done with some of the extensions to get them updated to work properly as well. 

1.6 introduced "Lazy Type Loading" which brought some significant performance improvements, but apparently at the cost of working well with different extensions. 

I've been working with @kidunot89 to get WPGraphQL for WooCommerce updated to be compatible and @justlevine released an update to WPGraphQL for Gravity forms (https://github.com/harness-software/wp-graphql-gravity-forms/pull/137) that brought it up to being compatible with 1.6+.

I'll post back when 1.6.2 is released, and if ya'll can test that and report back, that would be 👌🏻 

If things are _still_ broken after 1.6.2, I'd appreciate if you could try the ol' "de-activate plugins one at a time" trick and see which one is the issue where GraphiQL won't load, then we can start narrowing down and get to a resolution a bit quicker. 

If you determine that all other plugins work, but having WPGraphQL for ACF active is the issue, you can export your field groups as JSON and DM me in Slack so I can reproduce. 

If it's another extension that's causing the issue, post which one, and any relevant info to reproduce, and we can see what we can find out.

**Comment by jasonbahl** [comment]
Ooph. I don't think this should be closed quite yet. Re-opening.

**Comment by stevepepple** [comment]
Hope this update is helpful. I upgrade to 1.6.2, but still see the following issue in Gatsby + Wordpress: 

``` 
ERROR 

[gatsby-source-wordpress] Error category: undefined

Error:
  Internal server error

 Debug message:
  Return value of GraphQL\Type\Schema::resolveType() must be an instance of GraphQL\Type\Definition\Type, null returned

 Error path: __schema.types
```

**Comment by justlevine** [comment]
@stevepepple is that with all the other plugins disabled?

**Comment by vstrelianyi** [comment]
In my case the graphql plugin does not work with
- WPGraphQL WooCommerce (WooGraphQL)
- WP GraphQL Gutenberg
If I deactivate both of them - Graphql IDE load fine
otherwise Graphql IDE does not load and I have this error in console 
"invariant.mjs:6 Uncaught (in promise) Error: Invalid or incomplete introspection result. Ensure that you are passing "data" property of introspection response and no "errors" was returned alongside: undefined"

**Comment by eihsan94** [comment]
i checked ver 1.6.2
It still does not work with
- WP GraphQL Gutenberg

and version 1.6.2  does not work with my code snippets that i used to insert to extend graphql request posts
- code snippets
<pre>
function get_posts_by_slugs($slugs) {
	$posts = array();
	foreach ($slugs as &$value) {
		$the_slug = $value;
		$args = array(
		  'name'        => $the_slug,
		);
		$my_posts = get_posts($args);
		if( $my_posts ) :
			array_push($posts, $my_posts[0]);
		endif;
	}
	return $posts;
}

add_action( 'graphql_register_types', function() {
		$field_config = [
		  'type' => 'WP_Post',
		  'args' => [
			'myArg' => [
			  'type' => 'String',
			],
		  ],
		  'resolve' => function( $source, $args, $context, $info ) {
			if ( isset( $args['myArg'] ) ) {
				$the_slug = $args['myArg'];
				$args = array(
				  'name'        => $the_slug,
				);
				$my_posts = get_posts($args);
				return $my_posts[0];
			}
			return 'test';
		  },
		];
	register_graphql_field( 'ContentNode', 'getPostsBySlugs', $field_config);
} );
</pre>

**Comment by stevepepple** [comment]
> @stevepepple is that with all the other plugins disabled?

Disabling that all at once causes other issues, so kind of difficult to isolate. I'll test more, but suspect is also an issue with WP GraphQL Gutenberg or WPGraphQL for ACF.

**Comment by LeeBuckle** [comment]
Same issue here. For me, it seems to be related to WPGraphQL WooCommerce.

**When I have ACF/GQL + Woo/GQL active**

Console:
![Screenshot 2021-08-05 at 23 15 04](https://user-images.githubusercontent.com/80564456/128429892-569902d9-bd78-4769-b8b4-249ceff121e5.png)

Within ACF Field Group edit page:
![Screenshot 2021-08-05 at 23 26 13](https://user-images.githubusercontent.com/80564456/128430034-c71b4f72-083e-4ef8-a535-dd356d33cd3a.png)

When I deactivate WPGraphQL for Woocommerce, both of these errors go away and schema loads fine.

**Comment by jasonbahl** [comment]
👋🏻  We released an update to WPGraphQL for Gutenberg that is compatible with WPGraphQL v1.6. 

You can see the PR here: https://github.com/pristas-peter/wp-graphql-gutenberg/pull/117

And the release here: https://github.com/pristas-peter/wp-graphql-gutenberg/releases/tag/v0.3.9
And another bugfix released right after: https://github.com/pristas-peter/wp-graphql-gutenberg/releases/tag/v0.3.10

I'm still working with @kidunot89 to resolve issues with WPGraphQL for WooCommerce, but we're getting close!

**Comment by jasonbahl** [comment]
> Disabling that all at once causes other issues, so kind of difficult to isolate. I'll test more, but suspect is also an issue with WP GraphQL Gutenberg or WPGraphQL for ACF.

@stevepepple I understand disabling multiple plugins at once would cause issues with your client apps for sure, but in a staging environment you should be able to disable many or one at a time and test individual things, such as "Does GraphiQL Load the Schema"

If we're able to identify which plugin(s) cause the conflicts, then we can narrow down the specifics of the conflicts. 

----

So far I've had reports of issues with WPGraphQL v1.6 with: 

- WPGraphQL for ACF: (resolved by: https://github.com/wp-graphql/wp-graphql/pull/2051)
- WPGraphQL for Gutenberg: (resolved by: https://github.com/pristas-peter/wp-graphql-gutenberg/pull/117)
- WPGraphQL for Gravity Forms: (resolved by: https://github.com/harness-software/wp-graphql-gravity-forms/releases/tag/v0.7.3)
- WPGraphQL for WooCommerce: (I'm working with @kidunot89 on a resolution. . .getting close)
- WPGraphQL for Ninja Forms: (resolution progress, unknown)

**Comment by eihsan94** [comment]
I checked the 
wp-graphql v1.6.2 and wp-graphql-gutenbergv0.3.9
and i can confirm that it work just find
thank you so much for fixing this 🙌🏻

**Comment by PaperPlaneSoftware** [comment]
v1.6.2 is still broken for me

**Comment by jasonbahl** [comment]
@PaperPlaneSoftware can you export your ACF Fields as JSON and share them either here or via DM in Slack (you can join here: https://join.slack.com/t/wp-graphql/shared_invite/zt-3vloo60z-PpJV2PFIwEathWDOxCTTLA)

I'd love to see if I can reproduce.

**Comment by jasonbahl** [comment]
@PaperPlaneSoftware ah, you also said: 

> Thanks @jasonbahl. Here's a list of WPGraphQL plugins I'm using: Woocommerce, JWT and ACF

As stated in this comment (https://github.com/wp-graphql/wp-graphql/issues/2049#issuecomment-894549263) WPGraphQL for WooCommerce has known conflicts with WPGraphQL 1.6+ still. I'm working with @kidunot89 on getting it updated. Hopefully we're in a good state within a few days. 🤞🏻 

If you can de-activate WPGraphQL for WooCommerce, even momentarily, and see if GraphiQL loads again with your ACF Fields all in place, that would help eliminate ACF as the issue.

**Comment by PaperPlaneSoftware** [comment]
> If you can de-activate WPGraphQL for WooCommerce, even momentarily, and see if GraphiQL loads again with your ACF Fields all in place, that would help eliminate ACF as the issue.

Yup, it's WPGraphQL for WooCommerce that's doing it

**Comment by jasonbahl** [comment]
@PaperPlaneSoftware thanks for the confirmation! Hoping to have that update for WPGraphQL for WooCommerce shortly!

**Comment by jasonbahl** [comment]
@PaperPlaneSoftware WPGraphQL for WooCommerce v0.10.3 was released yesterday. Can you try updating and report back if things are working again for you? 🙏🏻

**Comment by PaperPlaneSoftware** [comment]
Yay! Looks like it's working now. Thanks @jasonbahl @kidunot89 👍

**Comment by LeeBuckle** [comment]
@jasonbahl seems to be working okay now. I've also checked the "Edit field group" in ACF which was displaying errors while GQL for Woo was active and the errors no longer appear there either. Thanks for the great work! 👍🏼

**Comment by calin-lucian** [comment]
Still getting an "ERROR Cannot read property 'getFields' of undefined" on gatsby Cloud. 

Using:  
WPGraphQL for Advanced Custom FieldsDeactivate - v0.5.3
WPGraphQL - v1.6.2

**Comment by bourneidBH** [comment]
> WPGraphQL for WooCommerce v0.10.3

I'm also having the same issue. Where would I get this version to test?

**Comment by jasonbahl** [comment]
@bourneidBH you can get latest WPGraphQL for WooCommerce here: https://github.com/wp-graphql/wp-graphql-woocommerce/releases

----

@calin-lucian if it's an issue specific to Gatsby, it _might_ be related to this issue: https://github.com/wp-graphql/wp-graphql/issues/2046

**Comment by mitchierichie** [comment]
I'm having this is as well.

**Comment by mitchierichie** [comment]
> Hope this update is helpful. I upgrade to 1.6.2, but still see the following issue in Gatsby + Wordpress:
> 
> ```
> ERROR 
> 
> [gatsby-source-wordpress] Error category: undefined
> 
> Error:
>   Internal server error
> 
>  Debug message:
>   Return value of GraphQL\Type\Schema::resolveType() must be an instance of GraphQL\Type\Definition\Type, null returned
> 
>  Error path: __schema.types
> ```

I'm in version 1.6.10 with no additional plugins or extensions enabled, still getting this error

**Comment by jasonbahl** [comment]
@mitchierichie what theme is active?

**Comment by mitchierichie** [comment]
> @mitchierichie what theme is active?

@jasonbahl Active theme is my custom child theme which is based on blank-canvas, which is actually also a child theme of seedlet.

**Comment by jasonbahl** [comment]
@mitchierichie if you change themes to something like twenty-sixteen, does the issue go away? I'm curious if there's some theme code that's causing a conflict?

**Comment by mitchierichie** [comment]
@jasonbahl I didn't have to change theme or anything, schema is working in 1.6.11 now

**Comment by jasonbahl** [comment]
@mitchierichie well that's awesome! Any chance you have the site on a domain like "site.com" but access the site so a domain like "site.wpengine.com" or "site.yourhost.com"? If so, this release had a fix for situations like that. If not, I'm glad things are fixed, but I'm curious what else in this released might have fixed things for you 😆

**Comment by mitchierichie** [comment]
> @mitchierichie well that's awesome! Any chance you have the site on a domain like "site.com" but access the site so a domain like "site.wpengine.com" or "site.yourhost.com"? If so, this release had a fix for situations like that. If not, I'm glad things are fixed, but I'm curious what else in this released might have fixed things for you 😆 
> 

My WordPress installation is on a subdomain `cms.mydomain.com`

Who knows if anything else was also affecting me

Thanks for the updates

**Comment by jasonbahl** [comment]
After reviewing all comments in this issue, I believe conflicts with WPGraphQL for WooCommerce and WPGraphQL for Gutenberg have all been addressed. 

There might be similar issues related to plugins that register settings using the `register_setting()` API, and that's addressed in PR #2262 

Based on that, I'm closing this issue.
